core/src/main/java/hudson/model/Failure.java

```diff
@@ -49,6 +49,11 @@ public Failure(String message) {
         this(message,false);
     }
 
+    public Failure(String message, Throwable cause) {
```

Comment:
```suggestion
    /**
     * @since TODO
     */
    public Failure(String message, Throwable cause) {
```

Reply:
Unsure about this change.

`Failure` is currently (slightly) inconsistent: `Descriptor.FormException` calls an overload of `#generateResponse` that takes a `Throwable` that will be set as request attribute. But why, given the class's Javadoc?

Now, if the idea is the Javadoc doesn't really apply anymore (see `AbstractModelObject/error.jelly` as evidence), would it not make more sense for us to change `#generateResponse` (the other one not taking a `Throwable`) to set the request property if there is a cause from this constructor? And if so, wouldn't we just deprecate `#generateResponse` taking a `Throwable`, and instead change callers to use the constructor taking a `Throwable`?

That would be a change in behavior however, as currently the caller being adapted in this PR doesn't render the exception, but would start doing so if `Failure` sets the `Throwable cause` from the constructor as request property. Perhaps we should have a flag to distinguish causes set for debugging diagnostics only, and causes intended to be rendered?

Probably easier to just skip this one for now.